src/yio-model/mediaplayer/epgmodel_mediaplayer.h

```diff
@@ -0,0 +1,186 @@
+ *****************************************************************************/
+
+#pragma once
+
+#include <QAbstractListModel>
+#include <QObject>
+#include <QVariant>
+#include <QtDebug>
+
+class EPGModelItem {
+ public:
+    EPGModelItem(const QString& key, const int& xCoordinate, const int& column,
+                       const int& width, const int& height, const QString& type, const QString& color,
+               const QString& title, const QString& subtitle, const QString& description,
+                 const QString& startTime, const QString& endTime, const QString& imageUrl, const QVariant& commands)
+        : m_key(key), m_xCoordinate(xCoordinate), m_column(column), m_width(width), m_height(height),
+          m_type(type), m_color(color), m_title(title), m_subtitle(subtitle), m_description(description),
+          m_startTime(startTime), m_endTime(endTime), m_imageUrl(imageUrl), m_commands(commands) {}
+
+    QString  itemKey() const { return m_key; }
+    int  itemXCoordinate() const { return m_xCoordinate; }
+    int  itemColumn() const { return m_column; }
+    int  itemWidth() const { return m_width; }
+    int  itemHeight() const { return m_height; }
+    QString  itemType() const { return m_type; }
+    QString  itemColor() const { return m_color; }
+    QString  itemTitle() const { return m_title; }
+    QString  itemSubTitle() const { return m_subtitle; }
+    QString  itemDescription() const { return m_description; }
+    QString  itemStartTime() const { return m_startTime; }
+    QString  itemEndTime() const { return m_endTime; }
+    QString  itemImageUrl() const { return m_imageUrl; }
+    QVariant itemCommands() const { return m_commands; }
+
+ private:
+    QString  m_key;
+    int  m_xCoordinate;
+    int  m_column;
+    int  m_width;
+    int m_height;
+    QString  m_type;
+    QString  m_color;
+    QString  m_title;
+    QString  m_subtitle;
+    QString  m_description;
+    QString  m_startTime;
+    QString  m_endTime;
+    QString  m_imageUrl;
+    QVariant m_commands;
+};
+
+class ListEPGModel : public QAbstractListModel {
+    Q_OBJECT
+    Q_PROPERTY(int count READ count WRITE setCount NOTIFY countChanged)
+
+ public:
+    enum SearchRoles { KeyRole = Qt::UserRole + 1, XCoordinateRole, ColumnRole, WidthRole,
+                       HeightRole, TypeRole, ColorRole, AnchorleftRole, TitleRole, SubTitleRole,
+                       DescriptionRole, StartTimeRole, EndTimeRole, ImageUrlRole, CommandsRole };
+
+    explicit ListEPGModel(QObject* parent = nullptr);
+    ~ListEPGModel() {}
+
+    int                    count() const;
+    int                    rowCount(const QModelIndex& p = QModelIndex()) const;
+    QVariant               data(const QModelIndex& index, int role = Qt::DisplayRole) const;
+    QHash<int, QByteArray> roleNames() const;
+
+    void append(const EPGModelItem& o);
+
+ public slots:  // NOLINT open issue: https://github.com/cpplint/cpplint/pull/99
```

Comment:
NOLINT is finally no longer required since yesterday ;-)

Reply:
So it's removed